lib/script.ex

```diff
@@ -0,0 +1,669 @@
+defmodule Bitcoinex.Script do
+  @moduledoc """
+  	a module for manipulating Bitcoin Scripts
+  """
+
+  import Bitcoinex.Opcode
+
+  alias Bitcoinex.Secp256k1.Point
+
+  alias Bitcoinex.{Utils, Address, Segwit, Base58, Network}
+
+  @wsh_length 32
+  @tapkey_length 32
+  @h160_length 20
+
+  @type script_type :: :p2pk | :p2pkh | :p2sh | :p2wpkh | :p2wsh | :p2tr | :non_standard
+
+  @type t :: %__MODULE__{
+          items: list
+        }
+
+  @enforce_keys [
+    :items
+  ]
+  defstruct [:items]
+
+  defp invalid_opcode_error(msg), do: {:error, "invalid opcode: #{msg}"}
+
+  def is_valid_opcode(i) when is_integer(i), do: i >= 0x00 && i < 0xFF
+
+  @doc """
+  	new returns an empty script object.
+  """
+  @spec new() :: t()
+  def new, do: %__MODULE__{items: []}
+
+  @doc """
+  	is_true? returns true if the script's only item is op_true, false otherwise.
```

Comment:
This function is confusing to me. I think we should instead have a separate evaluate function(or even in new Evaluator module). Or there is some other use case I overlook?

Reply:
No, you are correct. this is more for evaluate. I'll remove it.